File: python/215_kth_largest_element_in_an_array.py
import heapq

class Solution(object):
    def findKthLargest(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: int
        """
        
        heap = []
        for i in range(min(len(nums), k)):
            heapq.heappush(heap, nums[i])
        for i in range(k, len(nums)):
            heapq.heappushpop(heap, nums[i])

        return heap[0]

File: cpp/215_kth_largest_element_in_an_array.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    int findKthLargest(vector<int>& nums, int k) {

		priority_queue<int, vector<int>, greater<int>> q;
		for (int x : nums) {
			q.push(x);
			if (q.size() > k) {
				q.pop();
			}
		}
		return q.top();

    }
};
